lib/index.js

```diff
@@ -12,34 +26,308 @@ class PluginPassportOAuth {
+   *
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  getCredentialsFromUserId (userId, provider) {
+    return this.getProviderRepository(provider).search({
+      query: {
+        match: {
+          userId
+        }
+      }
+    })
+      .then(result => {
+        if (result.total === 0) {
+          return Promise.resolve(null);
+        }
+
+        return Promise.resolve(result.hits[0]);
+      });
+  }
+
+  /**
+   * Verify function called by passport to log the user.
+   * If a persist option is in the configuration of the strategy then every field present in this configuration will be persisted in the repository.
+   *
+   * @param request
+   * @param accessToken
+   * @param refreshToken
+   * @param profile
+   * @returns {Promise.<T>}
+   */
+  verify (request, accessToken, refreshToken, profile) {
+    const
+      user = {};
+
+    return this.getProviderRepository(profile.provider)
+      .get(profile._json[this.config.strategies[profile.provider].identifierAttribute])
+      .then(userObject => {
+        let req = null;
+
+        if (userObject !== null) {
+          return Promise.resolve(userObject.userId);
+        }
+
+        Object.keys(profile._json).forEach(attr => {
+          if (this.config.strategies[profile.provider].persist.indexOf(attr) > -1) {
+            user[attr] = profile._json[attr];
+          }
+        });
+
+        if (this.config.strategies[profile.provider].persist.length !== 0) {
+          req = {
+            controller: 'security',
+            action: 'createUser',
+            _id: profile._json[this.config.strategies[profile.provider].identifierAttribute],
+            body: {
+              content: {
+                profileIds: this.config.defaultProfiles ? this.config.defaultProfiles : ['default']
+              }, // content will be persisted in Kuzzle
+              credentials: {} // credentials will be persisted in the repository
+            }
+          };
+
+          if (this.config.strategies[profile.provider].kuzzleAttributesMapping) {
+            Object.keys(this.config.strategies[profile.provider].kuzzleAttributesMapping)
+              .forEach(attr => {
+                req.body.content[attr] = profile._json[this.config.strategies[profile.provider].kuzzleAttributesMapping[attr]];
+              });
+          }
+
+          user._id = profile._json[this.config.strategies[profile.provider].identifierAttribute];
+          req.body.credentials[profile.provider] = user;
+
+          return this.context.accessors.execute(new this.context.constructors.Request(request.original, req))
+            .then(() => {
+              return Promise.resolve(profile._json[this.config.strategies[profile.provider].identifierAttribute]);
+            });
+        }
+        return Promise.resolve(profile._json[this.config.strategies[profile.provider].identifierAttribute]);
```

Comment:
I don't agree with this. We could login from an external service and still have a volatile JWT token from Kuzzle, I think this is a good feature to have. We must discuss it.